keras_nlp/samplers/greedy_sampler.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Greedy Sampler."""
+
+import tensorflow as tf
+
+from keras_nlp.samplers.sampler import Sampler
+from keras_nlp.samplers.sampler import base_sampler_keyword_args
+from keras_nlp.samplers.sampler import call_keyword_docstring
+from keras_nlp.samplers.sampler import sample_keyword_docstring
+
+
+class GreedySampler(Sampler):
+    """Greedy Sampler class.
+
+    This sampler is implemented on greedy search, i.e., always picking up the
+    token of the largest probability as the next token.
+
+    Args:
+        {{base_sampler_keyword_args}}
+
+    Call Args:
+        {{call_keyword_args}}
+
+    Examples:
+    ```python
+    BATCH_SIZE = 8
```

Comment:
I don't see most of these params used more than once so you can inline them to match our other docstrings.

Reply:
This is just one style - some people prefer to define macros in one place, this is the same as our util docstring, e.g., [link](https://github.com/keras-team/keras-nlp/blob/eb764802d8accd950bb9bf4f0355f6ac1524024d/keras_nlp/utils/text_generation.py#L131)

Reply:
`utils.text_generation` has a lot of style weirdness I've been cleaning up. I'd prefer the args inlined

Reply:
I got your idea! I am open to both solutions, sharing my opinion here: to me it's a bit different - defining all constants/hypers before usage is cleaner to me, like in our guide: https://keras.io/examples/nlp/neural_machine_translation_with_keras_nlp/#setup, where `EPOCHS` is only used once.